add 'depend' make target to install dependencies

dmitris opened the pull request:
PR corrects the installation instructions - currently install.txt and install-linux.txt as well as https://github.com/returntocorp/pfff/wiki say to run `make depend`, but this target is not or no longer present in the Makefile.  `make opt` is also unknown.
The PR add the `depend` target with `opam install . --deps-only` and removes `make opt`.  I believe we should remove `make opt` from the Wiki page.

https://github.com/returntocorp/pfff/blob/develop/install.txt#L12
https://github.com/returntocorp/pfff/blob/develop/install_linux.txt#L9

dmitris committed:
remove 'depend' make target due to dune usage

aryx commented on Makefile:
```diff
@@ -1,5 +1,7 @@
 all:
 	dune build
+depend:
```
We needed depend when we were using Makefiles to build pfff, but now that we use dune,
the dependencies are managed automatically so you can remove this target
(installing the dependencies of the project is another thing; it should not be confused with what
most OCaml programmers expect from a 'make depend').

aryx commented on install_linux.txt:
```diff
@@ -8,4 +8,4 @@ Then:
   $ ./configure
   $ make depend
```
you can remove this line; it's not useful anymore.

aryx reviewed (changes requested):
Thx for doing that!

dmitris commented:
removed `make depend` in 8e95f8d

dmitris commented:
@aryx should we remove this `#note` line as well? https://github.com/returntocorp/pfff/blob/develop/commons/Makefile#L8

With this PR version, here are the remaining hits on `make depend`:
```
$ rg 'make depend'
Makefile.common
125:# but if your 'make depend' is wrong and include directories under OPAM,

commons/Makefile
8:#note: if you add a file (a .mli or .ml), dont forget to 'make depend'

commons/Makefile.common
117:# but if your 'make depend' is wrong and include directories under OPAM,

commons_wrappers/Makefile
9:# note: if you add a file (a .mli or .ml), dont forget to redo a 'make depend'

scripts/build
15:#make depend

docs/manual/announce_codemap_ocaml_list.txt
27: $ make depend; make; make opt

docs/manual/announce_treemap_viewer.txt
21:$ make depend; make

lang_ruby/parsing/Makefile
83:# but in make depend the .cmo are not there yet
89:	make depend
```

dmitris committed:
remote 'make depend' from configure's output

aryx commented:
nah, you can leave the other mention of depend.
I should remove all those Makefile at some point because we switched to the dune files, but I keep them just in case.
What matters are whether the instructions are up to date (which you fixed).

dmitris commented:
I'd like to mention that you need to run `opam install . --deps-only` before running `make`, otherwise when following installation instructions, I'm getting the following:
```
All seems fine for pfff.
To compile pfff type:
  $ make
To test pfff simply type:
  $ ./pfff -dump_php demos/foo.php

If you want to use the code visualization tools see the readme.txt file.

[dsavints@sdlcbuild01 pfff]$ make
dune build
File "commons/dune", line 5, characters 10-29:
5 |    yojson easy_logging_yojson
              ^^^^^^^^^^^^^^^^^^^
Error: Library "easy_logging_yojson" not found.
Hint: try:
  dune external-lib-deps --missing @@default
File "commons_core/dune", line 5, characters 3-15:
5 |    ANSITerminal
       ^^^^^^^^^^^^
Error: Library "ANSITerminal" not found.
Hint: try:
  dune external-lib-deps --missing @@default
File "commons_wrappers/graph/dune", line 6, characters 3-13:
6 |    ocamlgraph
       ^^^^^^^^^^
Error: Library "ocamlgraph" not found.
Hint: try:
  dune external-lib-deps --missing @@default
File "h_program-lang/dune", line 11, characters 18-35:
11 |  (preprocess (pps ppx_deriving.show))
                       ^^^^^^^^^^^^^^^^^
Error: Library "ppx_deriving.show" not found.
Hint: try:
  dune external-lib-deps --missing @@default
File "h_visualization/dune", line 11, characters 18-35:
11 |  (preprocess (pps ppx_deriving.show))
                       ^^^^^^^^^^^^^^^^^
[...]
```

dmitris commented:
do you think the following would be OK for `install.txt`:
```
To install pfff, you must first install recent versions of:
 - OCaml (at least 4.02.3)
   see http://caml.inria.fr/download.en.html
 - opam - http://opam.ocaml.org
 - dune - `opam install dune`, https://opam.ocaml.org/packages/dune/
 - SWI-prolog
   if you want the interactive code query tool
 - Java (for bddbddb)
   if you want the datalog pointer analysis

If you want most of the good things just type:

  $ ./configure
  $ opam install . --deps-only
  $ make

This should generate a few binaries in `_build/install/default/bin/` such
as pfff.  To install binaries and libraries, run `make install`.
```

aryx commented:
Yep sounds good. Actually pfff.opam seems to say you need OCaml 4.07.0 at least.

dmitris committed:
update install instructions, ocaml min 4.07.0

dmitris commented:
updated.  I changed `./pfff -parse_php demos/foo.php` in README.md to `pfff [....]` since the `pfff` executable is no longer in the root directory, so maybe we can rely on `make install` and `~/.opam/4.11.1/bin` being in the PATH.

What about the `Usage for pfff_db` section in README.md - does `pfff_db` still exist, or should we remove that section?

dmitris commented:
@aryx could you also update https://github.com/returntocorp/pfff/wiki#install, please? Maybe just put the following there:
```
  $ ./configure
  $ opam install . --deps-only
  $ make
```

aryx commented:
I've updated the wiki. Thx @dmitris for the suggestion and fix.
